src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -194,10 +196,29 @@ private string GetRootPathWithGuid_NoLock()
             var fullName = GetFullReflectionName(containingOrThis);
 
             var compilation = await temporaryDocument.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
-            // TODO: retrieve path to actual assembly instead of reference assembly
-            var reference = compilation.GetMetadataReference(symbol.ContainingAssembly);
+
+            string assemblyLocation = null;
+            var isReferenceAssembly = symbol.ContainingAssembly.GetAttributes().Any(attribute => attribute.AttributeClass.Name == nameof(ReferenceAssemblyAttribute));
+            if (isReferenceAssembly)
+            {
+                try
+                {
+                    var fullAssemblyName = symbol.ContainingAssembly.Identity.GetDisplayName();
+                    GlobalAssemblyCache.Instance.ResolvePartialName(fullAssemblyName, out assemblyLocation, preferredCulture: CultureInfo.CurrentCulture);
+                }
+                catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
+                {
+                }
+            }
+
+            if (assemblyLocation == null)
+            {
+                var reference = compilation.GetMetadataReference(symbol.ContainingAssembly);
+                assemblyLocation = reference.Display;
```

Comment:
> reference.Display [](start = 34, length = 18)

`Display` is only meant to be used in messages for the user. 
To get the path of the assembly use `(reference as PortableExecutableReference peReference)?.FilePath` and handle case when the result is null.

Reply:
> handle case when the result is null

What would an appropriate error handling strategy be here?

Reply:
I guess we should report that we can't navigate to the symbol. 

---
In reply to: [165761856](https://github.com/dotnet/roslyn/pull/24291#discussion_r165761856) [](ancestors = 165761856)